packages/scripts/src/lib/_migrate-glintrc.ts

```diff
@@ -0,0 +1,302 @@
+  });
+  if (errors.length) {
+    return err(`Could not parse YAML:\n\t${errors.join('\n\t')}`);
+  }
+
+  return tryOrElse(
+    (e) => {
+      assert(e instanceof z.ZodError, 'error somehow not a zod error');
+      return `Could not parse data as a Glint config:\n\t${e.format()._errors.join('\n\t')}`;
+    },
+    () => {
+      return GlintRc.parse(yamlData);
+    }
+  );
+}
+
+// Represents the part of a tsconfig.json file we care about. That is: not much!
+type ReadyToTransform = {
+  glint?: GlintConfig;
+};
+
+function loadTsconfig(contents: unknown): Result<ReadyToTransform, string> {
+  if (typeof contents !== 'string') {
+    return err(`Could not patch ${JSON.stringify(contents)}: not a string`);
+  }
+
+  return tryOrElse(
+    (e) => `Could not patch data:\n\t${e instanceof Error ? e.message : JSON.stringify(e)}`,
+    () => {
+      let result = evaluate(contents);
+
+      if (typeof result !== 'object' || result == null || Array.isArray(result)) {
+        throw new Error('invalid contents of tsconfig.json file');
+      }
+
+      return result as ReadyToTransform;
+    }
+  );
+}
+
+function patchTsconfig(contents: unknown, rc: GlintRc): Result<string, string> {
+  if (typeof contents !== 'string') {
+    return err(`Could not patch ${JSON.stringify(contents)}: not a string`);
+  }
+
+  return loadTsconfig(contents)
+    .andThen<ReadyToTransform>((loadedTsconfig) =>
+      loadedTsconfig.glint
+        ? err('There is already a glint config present, not overriding it')
+        : ok(loadedTsconfig)
+    )
+    .andThen((config) =>
+      tryOrElse(
+        (e) => {
+          return `Could not patch data:\n\t${JSON.stringify(e)}`;
+        },
+        () => {
+          // Build up the modified config incrementally, *not* adding
+          // `undefined` to it at any point, because we cannot patch in
+          // `undefined`!
+          config.glint = {};
+
+          if (rc.environment !== undefined) {
+            config.glint.environment = rc.environment;
+          }
+
+          if (rc.checkStandaloneTemplates !== undefined) {
+            config.glint.checkStandaloneTemplates = rc.checkStandaloneTemplates;
+          }
+
+          if (rc.include || rc.exclude) {
+            config.glint.transform = {};
+
+            if (rc.include !== undefined) {
+              config.glint.transform.include = rc.include;
+            }
+
+            if (rc.exclude !== undefined) {
+              config.glint.transform.exclude = rc.exclude;
+            }
+          }
+
+          return patch(contents, config);
+        }
+      )
+    );
+}
+
+function settledToResult<T>(
+  settledResults: Array<PromiseSettledResult<Result<T, unknown>>>
+): Array<Result<T, string>>;
+function settledToResult<T>(
+  settledResults: Array<PromiseSettledResult<T>>
+): Array<Result<T, string>>;
+function settledToResult<T>(
+  settledResults: Array<PromiseSettledResult<T | Result<T, unknown>>>
+): Array<Result<T, string>> {
```

Comment:
Needs overrides because otherwise it'll happily resolve an input of `Array<PromiseSettledResult<T>>` where `T` is `Result<A, B>` to an output of `Array<Result<Result<A, B>, string>>`. 😑